core/services/keystore/keys/ocr2key/key_bundle.go

```diff
@@ -71,28 +54,69 @@ func (ekb *EncryptedKeyBundle) SetID(value string) error {
 	}
 
 	copy(result[:], decodedString[:32])
-	ekb.ID = result
+	kb.id = result
 	return nil
 }
 
-// NewKeyBundle makes a new set of OCR key bundles from cryptographically secure entropy
-func NewKeyBundle() (*KeyBundle, error) {
-	return newKeyBundleFrom(cryptorand.Reader, cryptorand.Reader, cryptorand.Reader)
+// New makes a new set of OCR key bundles from cryptographically secure entropy
+func New(chainType chaintype.ChainType) (*KeyBundle, error) {
+	return newKeyBundleFrom(chainType, cryptorand.Reader, cryptorand.Reader, cryptorand.Reader)
 }
 
-func newKeyBundleFrom(onchainSigningKeyMaterial, onchainEncryptionKeyMaterial, offchainKeyMaterial io.Reader) (*KeyBundle, error) {
-	onchainKeyring, err := newEthereumKeyring(offchainKeyMaterial)
+func NewFromOCR1Key(v1key ocrkey.KeyV2) (KeyBundle, error) {
+	evmKeyring := EVMKeyring{
+		privateKey: ecdsa.PrivateKey(*v1key.OnChainSigning),
+	}
+	offChainKeyRing := OffchainKeyring{
+		signingKey:    ed25519.PrivateKey(*v1key.OffChainSigning),
+		encryptionKey: *v1key.OffChainEncryption,
+	}
+	k := KeyBundle{
+		ChainType:       chaintype.EVM,
+		evmKeyring:      evmKeyring,
+		OffchainKeyring: offChainKeyRing,
+	}
+	marshalledPrivK, err := k.Marshal()
 	if err != nil {
-		return nil, err
+		return KeyBundle{}, err
 	}
+	k.id = sha256.Sum256(marshalledPrivK)
+	return k, nil
+}
+
+func MustNewInsecure(reader io.Reader, chainType chaintype.ChainType) KeyBundle {
+	key, err := newKeyBundleFrom(chainType, reader, reader, reader)
+	if err != nil {
+		panic(errors.Wrap(err, "failed to generate new OCR2 Key"))
+	}
+	return *key
+}
+
+func newKeyBundleFrom(chainType chaintype.ChainType, onchainSigningKeyMaterial, onchainEncryptionKeyMaterial, offchainKeyMaterial io.Reader) (*KeyBundle, error) {
 	offchainKeyring, err := newOffchainKeyring(onchainSigningKeyMaterial, onchainEncryptionKeyMaterial)
 	if err != nil {
 		return nil, err
 	}
 	k := &KeyBundle{
-		OnchainKeyring:  *onchainKeyring,
+		ChainType:       chainType,
 		OffchainKeyring: *offchainKeyring,
 	}
+	switch chainType {
```

Comment:
I think you can avoid all these switches by just defining a keybundle interface like:
```

type KeyBundle interface {
	ocrtypes.OnchainKeyring
	ocrtypes.OffchainKeyring
	ChainType() chaintype.ChainType
	Marshal() ([]byte, error)
	PublicKeyAddressOnChain() string
	Unmarshal(b []byte) (err error)
}
```

implemented by a solanaKeyBundle and an evmKeyBundle